loadgen/logging.cc

```diff
@@ -421,7 +417,8 @@ PerfClock::time_point AsyncLog::GetMaxCompletionTime() {
 }
 
 QuerySampleLatency AsyncLog::GetMaxLatencySoFar() {
-  return max_latency_.load(std::memory_order_release);
+  std::unique_lock<std::mutex> lock(latencies_mutex_);
```

Comment:
Just curious: why does this need to hold mutex?

Reply:
If `GetMaxLatencySoFar` were called in one thread and `RecordSampleCompletion` were called concurrently in another thread, there would be a race. Looking at the surrounding code, I don't think that is possible, but if a variable is guarded by a mutex in one context, I think it ought to have an equivalent guard in all contexts to be safe. This function isn't performance critical, so I'm not worried about grabbing a lock here, particularly since I am pretty certain there will be no contention.

Reply:
makes sense